src/track/serato/beatgrid.cpp

```diff
@@ -458,7 +458,7 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
         }
     }
 
-    nonTerminalMarkers.reserve(markers.size());
+    nonTerminalMarkers.reserve(static_cast<int>(markers.size()));
```

Comment:
```suggestion
    nonTerminalMarkers.reserve(static_cast<qsizetype>(markers.size()));
```
To remain warning free with Qt6. 

Please be careful in general when casting size-related types to `int`.

Reply:
nonTerminalMarkers is a QList in Qt5 it has [reserve(int alloc)](https://doc.qt.io/qt-5/qlist.html#reserve)
A qsizetype cast is a std::ptrdiff_t in Qt5 and Qt6 so the proposed change will become a lossy operation in Qt5.
So for now the current solution works warning free.   
We may consider to introduce a type that is qsizetype in Qt 6 and int in Qt 5 to make it correct. But IMHO there is no technical need and that will be surprising cast for all new contributors.

Reply:
Mhmm yeah, thats unfortunate. Those warnings will come back for Qt6 though since there `reserve` takes a `qsizetype` (I was hoping it was just `int` on Qt5). Introducing ifdefs everywhere is also bad. I guess its better to have a widening conversion than narrowing... so I guess we can keep `int`.